### ISSUE TITLE
Improve "Analyzer [my_analyzer] must have a type associated with it" error

### ISSUE THREAD

**Issue by javanna** [elastic/elasticsearch#15196]
Trying to create a custom analyzer as part of index settings, while creating an index, if you forget to specify the "tokenizer" (or you misspell tokenizer) you get back an "Analyzer [my_analyzer] must have a type associated with it" error, which seems to be the wrong error message. It should rather say that the tokenizer is missing.

```
PUT test
{
  "settings" : {
    "analysis": {
      "analyzer": {
        "my_analyzer" : {
          "tokenizer_misspelled" : "whitespace",
          "filter" : ["lowercase", "stop", "snowball"]
        }
      }
    }
  }
}
```

The fact that a misspelled key is accepted although it will get ignored relates to #15183

**Comment by clintongormley** [comment]
An analyzer can be based on another analyzer (by specifying `type`) or it defaults to being a `type: custom` analyzer if you specify `tokenizer`.  I'm not sure that we want to require an explicit `type: custom` line for every analyzer, but the error message could say: "An analyzer must either have a `type` or a `tokenizer` parameter"

**Comment by javanna** [comment]
I see that same error in a couple of places in `AnalysisRegistry`. One occurrence relates to the specific case when "tokenizer" is not there, so I think we can update that error safely without affecting the other one that relates to "type". https://github.com/elastic/elasticsearch/blob/master/core/src/main/java/org/elasticsearch/index/analysis/AnalysisRegistry.java#L312

**Comment by johtani** [comment]
In 5.0.0-alpha4, I got "analyzer [my_analyzer] must specify either an analyzer type, or a tokenizer" as an error message.
I think this already closed by #18455
